libnavigation-core/api/current.txt

```diff
@@ -83,6 +83,7 @@ package com.mapbox.navigation.core.arrival {
   public interface ArrivalObserver {
     method public void onFinalDestinationArrival(com.mapbox.navigation.base.trip.model.RouteProgress routeProgress);
     method public void onNextRouteLegStart(com.mapbox.navigation.base.trip.model.RouteLegProgress routeLegProgress);
+    method public void onWaypointArrival(com.mapbox.navigation.base.trip.model.RouteProgress routeProgress);
```

Comment:
Should it be for semantics consistency? 🤔 

```suggestion
    method public void onRouteLegArrival(com.mapbox.navigation.base.trip.model.RouteProgress routeProgress);
```

Reply:
Preferring waypoint to be consistent with the [directions api waypoint](https://docs.mapbox.com/api/navigation/directions/#waypoint-object). Arriving at a Waypoint.

The consistency was influenced by `onFinalDestinationArrival`.
it was  [onRouteArrival](https://github.com/mapbox/mapbox-navigation-android/commit/e92d4b77a4d7f99a767dd6c6c49179d57486f377#diff-99ddd5cade177822f9f38a9174b691724247f56161732e7c39ba1876313a146bL24-R24) but that was considered confusing, but would be more consistent with onRouteLegArrival

Personal preference is consistency with the [directions api waypoints](https://github.com/mapbox/mapbox-java/blob/73e171930c6be2bd44b7d725daf3bde9e819c9b7/services-directions/src/main/java/com/mapbox/api/directions/v5/MapboxDirections.java#L289-L296). Actually second guessing whether this is accurate then. Does WaypointArrival become confusing with Silent Waypoints?